addons/website_forum/controllers/website_forum.py

```diff
@@ -670,7 +675,7 @@ def _prepare_open_forum_user(self, user, forums, **kwargs):
         # questions and answers by user
         user_question_ids = Post.search([
             ('parent_id', '=', False),
-            ('forum_id', 'in', forums.ids), ('create_uid', '=', user.id)],
+            ('forum_id', 'in', forums.ids), ('create_uid', '=', user.id), ('can_view', '=', True)],
```

Comment:
I wonder if we should not just add `can_view` in the access rules directly (it's searchable so it should work)